packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -562,14 +552,17 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
                   deps.size > 1 ||
                   // main chunk is removed
                   (hasRemovedPureCssChunk && deps.size > 0)
-                    ? [...deps]
+                    ? modulePreload === false
+                      ? [...deps].filter((d) => d.endsWith('.css'))
+                      : [...deps]
```

Comment:
It would be nice to add a comment here why we're still preloading for CSS files specifically.

Reply:
done here: [chore: add comment](https://github.com/vitejs/vite/pull/13973/commits/e834315f23f3630701c5ba557344ead9b600d9e7)